test/testScripts/providers/hoverTest.test.ts

```diff
@@ -0,0 +1,181 @@
+import { expect } from 'chai';
+import { createTestWorkspaceManager, getDoc } from '../../utils/helper';
+import { doHover } from '../../../src/providers/hoverProvider';
+import { Position } from 'vscode-languageserver';
+
+describe('doHover()', () => {
+  const workspaceManager = createTestWorkspaceManager();
+
+  describe('Play keywords hover', () => {
+    const tests = [
+      {
+        word: 'name',
+        position: { line: 0, character: 4 } as Position,
+        doc: 'Identifier. Can be used for documentation, or in tasks/handlers.',
+      },
+      {
+        word: 'host',
+        position: { line: 1, character: 4 } as Position,
+        doc: 'A list of groups, hosts or host pattern that translates into a list of hosts that are the play’s target.',
+      },
+      {
+        word: 'tasks',
+        position: { line: 3, character: 4 } as Position,
+        doc: 'Main list of tasks to execute in the play, they run after roles and before post_tasks.',
+      },
+    ];
+
+    tests.forEach(({ word, position, doc }) => {
+      it(`should provide hovering for '${word}'`, async function () {
+        const textDoc = await getDoc('hover/tasks.yml');
+        const context = workspaceManager.getContext(textDoc.uri);
```

Comment:
Can these be moved outside of for-each? It doesn't seem to be necessary to repeat the same hardcoded non-parametrized action again and again.

Reply:
This cannot be moved outside because mocha `describe` doesn't support async code and we thus we have to put the async code inside mocha `it`. That's the reason why it is done this way.